corehq/motech/views.py

```diff
@@ -223,3 +228,42 @@ def form_valid(self, form):
         return super().form_valid(form)
 
 
+@require_POST
+@require_permission(Permissions.edit_motech)
+@requires_privilege_with_fallback(privileges.DATA_FORWARDING)
+def test_connection_settings(request, domain):
+    if request.POST.get('plaintext_password') == PASSWORD_PLACEHOLDER:
+        # The user is editing an existing instance, and the form is
+        # showing the password placeholder. (We don't tell the user what
+        # the API password is.)
+        return JsonResponse({
+            "success": False,
+            "response": _("Please enter API password again."),
+        })
```

Comment:
Would it make sense to move this into the form validation logic, or should that not know about this stuff?

Reply:
It's tricky.

We use `PASSWORD_PLACEHOLDER` (which is `'*' * 16`) to indicate to the user that the password has been set. (We do not send the actual password to the user because they should not be using HQ as their password manager. :upside_down_face: )

So when they change values in this form and submit it, the form class must recognise `PASSWORD_PLACEHOLDER` as a valid value.

But when they want to test their connection, they will need to enter the real value, and so `PASSWORD_PLACEHOLDER` is not a valid value in this function.